Sample/sdk-ios-swift/ViewController.swift

```diff
@@ -11,127 +11,117 @@ import YotiButtonSDK
-                    // your own Service and handle the result, set  isProcessed at true.
-                    // Otherwise, set isProcessed at false and the Yoti SDK
-                    // will make the call.
-                    return false
-                })
-                .setCallbackBackendURL(url)
-                .setBackendCompletion({ (data, error) in
-                    print("=== Backend Completion ===")
-                    print("object: \(String(describing: data))")
-                    print("error: \(String(describing: error))")
-                    
-                    self.responseObject = try? JSONSerialization.jsonObject(with: data!, options: []) as! [String : Any]
-
-                    
-                    if error == nil {
-                        DispatchQueue.main.async {
-                            self.moveToProfile()
-                        }
-                    }
-                    
-                    
-                })
-                .create()
-            YotiSDK.add(scenario: scenario)
+    
+    @IBAction func yotiButtonDidTouchUpInside(_ sender: YotiButton) {
+        guard let useCaseID = sender.useCaseID else{
+             return
         }
-        catch {
-            
+        do {
+            UIApplication.shared.isNetworkActivityIndicatorVisible = true
+            try YotiSDK.startScenario(for: useCaseID, with: self)
+        } catch {
+            // FIXME
         }
-
-        // Do any additional setup after loading the view, typically from a nib.
     }
     
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        rememberMeButton.setTitle("RememberMe Scenario", for: .normal)
+        selfieAuthButton.setTitle("SelfieAuth Scenario", for: .normal)
+    }
+
     func moveToProfile() {
         performSegue(withIdentifier: "moveToProfile", sender: self)
     }
     
     override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-        
         guard let identifier = segue.identifier else {
             return
         }
-        
         switch identifier {
         case "moveToProfile":
             guard let viewController = segue.destination as? ProfileViewController else {
                 break
             }
-            
-            guard let responseObject = responseObject,
-                let userInfo = responseObject["userInfo"] as? [String: Any] else {
-                    break
+            guard let responseObject = responseObject, !responseObject.attributes.isEmpty else {
+                break
             }
-            
-            if let selfie = userInfo["selfie"] as? String
-            {
-                let base64Image = selfie.replacingOccurrences(of: "data:image/jpeg;base64,", with: "")
-                if let imageData = Data(base64Encoded: base64Image) {
-
-                    let photo = UIImage(data: imageData)
-                    viewController.selfie = photo
+            responseObject.attributes.forEach {
```

Comment:
it might be cleaner to have responseObject.attributes given to the next viewController and leave that one to fill itself.

Reply:
@CharlesVu : I am doing it in this ViewController because we are getting the JSON data here in this ViewController. We know about the structure of the JSON here. Passing the whole responseObject to the next ViewController would mean that I am assuming that this viewController needs to know about the structure of JSON.